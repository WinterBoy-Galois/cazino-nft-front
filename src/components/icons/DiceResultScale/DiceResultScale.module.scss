.cls-1 {
  fill: #081221;
}

.cls-4 {
  fill: #ffde00;
}

.cls-8 {
  fill: #22263a;
}

.cls-11 {
  fill: none;
  stroke: #0d1b31;
  stroke-miterlimit: 10;
  stroke-width: 1.89px;
}

.cls-12 {
  fill: #fff;
}

.text {
  fill: #fff;
  font-size: 60px;
  font-weight: 900;
}

.diamond {
  &__container {
    position: absolute;
    width: 15%;
    right: 24%;
    height: 72.6%;
    top: 8.1%;
  }
}
