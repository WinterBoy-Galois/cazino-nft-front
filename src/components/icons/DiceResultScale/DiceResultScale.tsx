import React, { Fragment, useMemo } from 'react';
import { appConfig } from '../../../common/config';
import { formatGameResult } from '../../../common/util/format.util';
import { useDimensions } from '../../../hooks/useDimensions.hook';
import Diamond from '../Diamond';
import styles from './DiceResultScale.module.scss';

interface IProps {
  result?: number;
}

const DiceResultScale = ({ result }: IProps) => {
  const { ref: containerRef, dimensions: containerDimensions } = useDimensions();
  const { ref: diamondRef, dimensions: diamondDimensions } = useDimensions();

  const diamondYPos = useMemo(() => {
    if (result) {
      return (containerDimensions.height ?? 0) * (1 - result / 100) - diamondDimensions.height / 2;
    }

    return (containerDimensions.height ?? 0) - diamondDimensions.height / 2;
  }, [containerDimensions, result, diamondDimensions]);

  return (
    <Fragment>
      <div className={styles.diamond__container} ref={containerRef}>
        <div ref={diamondRef}>
          <Diamond
            className={styles.diamond}
            style={{
              transform: `translateY(${diamondYPos}px)`,
              transition: `transform ${appConfig.diceGameDiamondTransitionTime}ms ease-out`,
            }}
          />
        </div>
      </div>

      <svg
        xmlns="http://www.w3.org/2000/svg"
        viewBox="0 0 616.94 669.15"
        style={{ width: '100%', height: '100%' }}
      >
        <g id="Layer_2" data-name="Layer 2">
          <g id="user_scale" data-name="user scale">
            <path
              className={styles['cls-1']}
              d="M270.52 35.02H489.8v24.02H270.52zM270.52 68.77H489.8v12.01H270.52z"
            />
            <path className={styles['cls-1']} d="M289.18 31.4h137.9v51.77h-137.9z" />
            <path fill="#2e475e" d="M411.92 22.72l16.81 537.36h61.07V12.52l-77.88 10.2z" />
            <path fill="#f39200" d="M616.94 614.58H244.82l3.36 34.76h368.76v-34.76z" />
            <path className={styles['cls-4']} d="M416.13 560.08h200.81v54.5h-199l-1.81-54.5z" />
            <path fill="#b66e00" d="M616.84 669.15H236.98l11.2-19.81h368.66v19.81z" />
            <path className={styles['cls-4']} d="M260.87 606.79h84.82v7.79h-84.82z" />
            <path fill="#be1622" d="M268.67 580.53h69.2v26.26h-69.2z" />
            <path fill="#cb454e" d="M313.5 580.53h13.91v26.26H313.5z" />
            <path
              className={styles['cls-8']}
              d="M261.39 210.83L22.75 190.15a4.88 4.88 0 01-4.43-4.3L0 28.69a4.9 4.9 0 014.43-5.43L268 0a4.88 4.88 0 015.31 5l-6.61 201.08a4.88 4.88 0 01-5.31 4.75z"
            />
            <path
              fill="#3e5a80"
              d="M22.91 185.56L4.56 27.86 268.72 4.57l-6.63 201.71-239.18-20.72z"
            />
            <path
              className={styles['cls-8']}
              d="M244.62 172.16l-208.24-4.27L23.4 56.33l226.48-30.24-5.26 146.07z"
            />
            <path
              fill="#0d1b31"
              d="M40.48 163.38L28.49 60.29l216.6-28.93-4.9 136.11-199.71-4.09z"
            />
            <text className={styles.text} x="10%" y="19%">
              {formatGameResult(result)}
            </text>
            <path className={styles['cls-1']} d="M289.18 26.09h23.32v66.43h-23.32z" />
            <path
              className={styles['cls-11']}
              d="M487.91 537.78h-42.76M487.91 528.07h-13.96M487.91 518.37h-13.96M487.91 508.67h-13.96M487.91 498.97h-13.96M487.91 489.27h-42.76M487.91 479.57h-13.96M487.91 469.86h-13.96M487.91 460.16h-13.96M487.91 450.46h-13.96M487.91 440.76h-42.76M487.91 431.06h-13.96M487.91 421.36h-13.96M487.91 411.65h-13.96M487.91 401.95h-13.96M487.91 392.25h-42.76M487.91 382.55h-13.96M487.91 372.85h-13.96M487.91 363.15h-13.96M487.91 353.44h-13.96M487.91 343.74h-42.76M487.91 334.04h-13.96M487.91 324.34h-13.96M487.91 314.64h-13.96M487.91 304.94h-13.96M487.91 295.23h-42.76M487.91 285.53h-13.96M487.91 275.83h-13.96M487.91 266.13h-13.96M487.91 256.43h-13.96M487.91 246.73h-42.76M487.91 237.02h-13.96M487.91 227.32h-13.96M487.91 217.62h-13.96M487.91 207.92h-13.96M487.91 198.22h-42.76M487.91 188.51h-13.96M487.91 178.81h-13.96M487.91 169.11h-13.96M487.91 159.41h-13.96M487.91 149.71h-42.76M487.91 140.01h-13.96M487.91 130.3h-13.96M487.91 120.6h-13.96M487.91 110.9h-13.96M487.91 101.2h-42.76M487.91 91.5h-13.96M487.91 81.8h-13.96M487.91 72.09h-13.96M487.91 62.39h-13.96M487.91 52.69h-42.76"
            />
            <path
              className={styles['cls-12']}
              d="M445.74 31.64h2.39a.29.29 0 01.22.09.25.25 0 01.08.21v16.9a.28.28 0 01-.08.21.29.29 0 01-.22.09h-2.29a.29.29 0 01-.22-.09.28.28 0 01-.08-.21v-14.3s0-.06-.05-.09a.11.11 0 00-.11 0l-1.77.52a.21.21 0 01-.12 0c-.15 0-.23-.09-.23-.28v-1.63a.37.37 0 01.23-.35l1.87-1a1.06 1.06 0 01.38-.07zM451.43 48.08a5 5 0 01-1.27-3.52v-8.35a4.91 4.91 0 011.27-3.51 5 5 0 016.79 0 4.84 4.84 0 011.29 3.51v8.35a4.88 4.88 0 01-1.29 3.52 5 5 0 01-6.79 0zm4.69-1.78a2.33 2.33 0 00.49-1.56V36a2.33 2.33 0 00-.49-1.56 1.58 1.58 0 00-1.31-.59 1.52 1.52 0 00-1.27.59 2.38 2.38 0 00-.48 1.56v8.7a2.38 2.38 0 00.48 1.56 1.55 1.55 0 001.27.59 1.61 1.61 0 001.31-.55zM462.18 48.08a5 5 0 01-1.27-3.52v-8.35a4.91 4.91 0 011.27-3.51 5 5 0 016.79 0 4.84 4.84 0 011.29 3.51v8.35a4.88 4.88 0 01-1.26 3.52 5 5 0 01-6.79 0zm4.69-1.78a2.33 2.33 0 00.49-1.56V36a2.33 2.33 0 00-.49-1.56 1.58 1.58 0 00-1.31-.59 1.52 1.52 0 00-1.27.59 2.38 2.38 0 00-.48 1.56v8.7a2.38 2.38 0 00.48 1.56 1.55 1.55 0 001.27.59 1.61 1.61 0 001.31-.55zM458.76 286.24a12.43 12.43 0 01-.1 1.75 3.84 3.84 0 01-1.3 2.85 4.62 4.62 0 01-3.12 1 4.5 4.5 0 01-3-.95 4.19 4.19 0 01-1.4-2.85l-.08-.7a.27.27 0 01.3-.3h2.3a.3.3 0 01.33.3 2.32 2.32 0 00.05.45c.2 1 .7 1.55 1.5 1.55a1.34 1.34 0 001-.4 2 2 0 00.51-1.12 10.08 10.08 0 00.1-1.6 7.44 7.44 0 00-.15-1.6 1.8 1.8 0 00-.51-1 1.36 1.36 0 00-1-.36 1.49 1.49 0 00-1.55 1.15.32.32 0 01-.33.28H450a.26.26 0 01-.21-.09.29.29 0 01-.09-.21v-9.93a.29.29 0 01.09-.21.26.26 0 01.21-.09h8.15a.31.31 0 01.3.3v1.9a.26.26 0 01-.09.21.29.29 0 01-.21.09h-5.43c-.08 0-.12 0-.12.13v4.87c0 .05 0 .08.05.09s.06 0 .1 0a3 3 0 011-.71 3.09 3.09 0 011.28-.26 3.19 3.19 0 012.42 1 4.48 4.48 0 011.08 2.71 10 10 0 01.23 1.75zM461.44 290.58a4.87 4.87 0 01-1.28-3.51v-8.35a4.91 4.91 0 011.28-3.52 4.44 4.44 0 013.37-1.33 4.53 4.53 0 013.42 1.33 4.91 4.91 0 011.28 3.52v8.35a4.87 4.87 0 01-1.28 3.51 4.54 4.54 0 01-3.42 1.34 4.44 4.44 0 01-3.37-1.34zm4.69-1.78a2.38 2.38 0 00.48-1.56v-8.7a2.38 2.38 0 00-.48-1.56 1.65 1.65 0 00-1.32-.59 1.55 1.55 0 00-1.27.59 2.38 2.38 0 00-.48 1.56v8.7a2.38 2.38 0 00.48 1.56 1.52 1.52 0 001.27.59 1.62 1.62 0 001.32-.59zM461.09 532.63a4.87 4.87 0 01-1.27-3.51v-8.35a4.87 4.87 0 011.27-3.51 4.47 4.47 0 013.38-1.34 4.52 4.52 0 013.41 1.34 4.84 4.84 0 011.29 3.51v8.35a4.84 4.84 0 01-1.29 3.51 4.48 4.48 0 01-3.41 1.34 4.43 4.43 0 01-3.38-1.34zm4.69-1.77a2.33 2.33 0 00.49-1.56v-8.7a2.38 2.38 0 00-.49-1.57 1.6 1.6 0 00-1.31-.58 1.57 1.57 0 00-1.28.58 2.49 2.49 0 00-.47 1.57v8.7a2.44 2.44 0 00.47 1.56 1.58 1.58 0 001.28.59 1.61 1.61 0 001.31-.59z"
            />
          </g>
        </g>
      </svg>
    </Fragment>
  );
};

export default DiceResultScale;
