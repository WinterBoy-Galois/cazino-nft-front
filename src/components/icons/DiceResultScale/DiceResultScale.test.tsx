import React from 'react';
import { render, cleanup } from '@testing-library/react';
import DiceResultScale from '.';

describe('<DiceResultScale />', () => {
  beforeEach(cleanup);

  it('renders correctly', () => {
    const { container } = render(<DiceResultScale />);
    expect(container).toMatchSnapshot();
  });
});
