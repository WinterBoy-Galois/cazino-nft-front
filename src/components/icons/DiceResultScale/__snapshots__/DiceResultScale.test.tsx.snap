// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DiceResultScale /> renders correctly 1`] = `
<div>
  <div
    class="diamond__container"
  >
    <div>
      <svg
        class="diamond"
        style="transform: translateY(0px); transition: transform 300ms ease-out;"
        viewBox="0 0 106.22 58.86"
        xmlns="http://www.w3.org/2000/svg"
      >
        <defs>
          <style>
            .cls-1{fill:#5cc2f3}.cls-2{fill:#8ddfff}.cls-3{fill:#1db4ef}.cls-5{fill:#fff}
          </style>
        </defs>
        <g
          data-name="Layer 2"
          id="Layer_2"
        >
          <g
            id="diamond"
          >
            <path
              class="cls-1"
              d="M106.22 27.02zM0 27.02zM43.84 0l62.38 27.02L43.84 0zM43.84 0L0 27.02 43.84 0zM43.84 54.05L0 27.02l43.84 27.03 62.38-27.03-62.38 27.03z"
            />
            <path
              class="cls-2"
              d="M43.84 27.02v27.03l62.38-27.03H43.84z"
            />
            <path
              class="cls-3"
              d="M43.84 27.02H0l43.84 27.03V27.02zM43.84 0v27.02h62.38L43.84 0z"
            />
            <path
              class="cls-2"
              d="M0 27.02h43.84V0L0 27.02z"
            />
            <path
              class="cls-1"
              d="M96.33 27.02L44.13 4.41 7.44 27.02h.01l36.68 22.62 52.2-22.62z"
            />
            <path
              class="cls-3"
              d="M43.89 4.56L7.53 26.97h36.36V4.56z"
            />
            <path
              class="cls-2"
              d="M7.53 26.97l-.08.05 36.44 22.47V26.97H7.53zM44.13 4.41l-.24.15v22.41h52.32L44.13 4.41z"
            />
            <path
              class="cls-3"
              d="M96.21 26.97H43.89v22.52l.24.15 52.2-22.62-.12-.05z"
            />
            <path
              class="cls-4"
              d="M16.97 27.02l27.56-16.99 39.22 16.99-39.22 16.99-27.56-16.99z"
            />
            <path
              class="cls-5"
              d="M25.67 10.15a6.5 6.5 0 006.07 6.5 6.5 6.5 0 00-6.07 6.51 6.51 6.51 0 00-6.08-6.51 6.51 6.51 0 006.08-6.5zM64.21 35.11A11.87 11.87 0 0075.28 47a11.87 11.87 0 00-11.07 11.86A11.89 11.89 0 0053.11 47a11.89 11.89 0 0011.1-11.89zM66.41 22.42a3.9 3.9 0 003.64 3.91 3.9 3.9 0 00-3.64 3.9 3.91 3.91 0 00-3.65-3.9 3.91 3.91 0 003.65-3.91z"
            />
          </g>
        </g>
      </svg>
    </div>
  </div>
  <svg
    style="width: 100%; height: 100%;"
    viewBox="0 0 616.94 669.15"
    xmlns="http://www.w3.org/2000/svg"
  >
    <g
      data-name="Layer 2"
      id="Layer_2"
    >
      <g
        data-name="user scale"
        id="user_scale"
      >
        <path
          class="cls-1"
          d="M270.52 35.02H489.8v24.02H270.52zM270.52 68.77H489.8v12.01H270.52z"
        />
        <path
          class="cls-1"
          d="M289.18 31.4h137.9v51.77h-137.9z"
        />
        <path
          d="M411.92 22.72l16.81 537.36h61.07V12.52l-77.88 10.2z"
          fill="#2e475e"
        />
        <path
          d="M616.94 614.58H244.82l3.36 34.76h368.76v-34.76z"
          fill="#f39200"
        />
        <path
          class="cls-4"
          d="M416.13 560.08h200.81v54.5h-199l-1.81-54.5z"
        />
        <path
          d="M616.84 669.15H236.98l11.2-19.81h368.66v19.81z"
          fill="#b66e00"
        />
        <path
          class="cls-4"
          d="M260.87 606.79h84.82v7.79h-84.82z"
        />
        <path
          d="M268.67 580.53h69.2v26.26h-69.2z"
          fill="#be1622"
        />
        <path
          d="M313.5 580.53h13.91v26.26H313.5z"
          fill="#cb454e"
        />
        <path
          class="cls-8"
          d="M261.39 210.83L22.75 190.15a4.88 4.88 0 01-4.43-4.3L0 28.69a4.9 4.9 0 014.43-5.43L268 0a4.88 4.88 0 015.31 5l-6.61 201.08a4.88 4.88 0 01-5.31 4.75z"
        />
        <path
          d="M22.91 185.56L4.56 27.86 268.72 4.57l-6.63 201.71-239.18-20.72z"
          fill="#3e5a80"
        />
        <path
          class="cls-8"
          d="M244.62 172.16l-208.24-4.27L23.4 56.33l226.48-30.24-5.26 146.07z"
        />
        <path
          d="M40.48 163.38L28.49 60.29l216.6-28.93-4.9 136.11-199.71-4.09z"
          fill="#0d1b31"
        />
        <text
          class="text"
          x="10%"
          y="19%"
        >
          —.—
        </text>
        <path
          class="cls-1"
          d="M289.18 26.09h23.32v66.43h-23.32z"
        />
        <path
          class="cls-11"
          d="M487.91 537.78h-42.76M487.91 528.07h-13.96M487.91 518.37h-13.96M487.91 508.67h-13.96M487.91 498.97h-13.96M487.91 489.27h-42.76M487.91 479.57h-13.96M487.91 469.86h-13.96M487.91 460.16h-13.96M487.91 450.46h-13.96M487.91 440.76h-42.76M487.91 431.06h-13.96M487.91 421.36h-13.96M487.91 411.65h-13.96M487.91 401.95h-13.96M487.91 392.25h-42.76M487.91 382.55h-13.96M487.91 372.85h-13.96M487.91 363.15h-13.96M487.91 353.44h-13.96M487.91 343.74h-42.76M487.91 334.04h-13.96M487.91 324.34h-13.96M487.91 314.64h-13.96M487.91 304.94h-13.96M487.91 295.23h-42.76M487.91 285.53h-13.96M487.91 275.83h-13.96M487.91 266.13h-13.96M487.91 256.43h-13.96M487.91 246.73h-42.76M487.91 237.02h-13.96M487.91 227.32h-13.96M487.91 217.62h-13.96M487.91 207.92h-13.96M487.91 198.22h-42.76M487.91 188.51h-13.96M487.91 178.81h-13.96M487.91 169.11h-13.96M487.91 159.41h-13.96M487.91 149.71h-42.76M487.91 140.01h-13.96M487.91 130.3h-13.96M487.91 120.6h-13.96M487.91 110.9h-13.96M487.91 101.2h-42.76M487.91 91.5h-13.96M487.91 81.8h-13.96M487.91 72.09h-13.96M487.91 62.39h-13.96M487.91 52.69h-42.76"
        />
        <path
          class="cls-12"
          d="M445.74 31.64h2.39a.29.29 0 01.22.09.25.25 0 01.08.21v16.9a.28.28 0 01-.08.21.29.29 0 01-.22.09h-2.29a.29.29 0 01-.22-.09.28.28 0 01-.08-.21v-14.3s0-.06-.05-.09a.11.11 0 00-.11 0l-1.77.52a.21.21 0 01-.12 0c-.15 0-.23-.09-.23-.28v-1.63a.37.37 0 01.23-.35l1.87-1a1.06 1.06 0 01.38-.07zM451.43 48.08a5 5 0 01-1.27-3.52v-8.35a4.91 4.91 0 011.27-3.51 5 5 0 016.79 0 4.84 4.84 0 011.29 3.51v8.35a4.88 4.88 0 01-1.29 3.52 5 5 0 01-6.79 0zm4.69-1.78a2.33 2.33 0 00.49-1.56V36a2.33 2.33 0 00-.49-1.56 1.58 1.58 0 00-1.31-.59 1.52 1.52 0 00-1.27.59 2.38 2.38 0 00-.48 1.56v8.7a2.38 2.38 0 00.48 1.56 1.55 1.55 0 001.27.59 1.61 1.61 0 001.31-.55zM462.18 48.08a5 5 0 01-1.27-3.52v-8.35a4.91 4.91 0 011.27-3.51 5 5 0 016.79 0 4.84 4.84 0 011.29 3.51v8.35a4.88 4.88 0 01-1.26 3.52 5 5 0 01-6.79 0zm4.69-1.78a2.33 2.33 0 00.49-1.56V36a2.33 2.33 0 00-.49-1.56 1.58 1.58 0 00-1.31-.59 1.52 1.52 0 00-1.27.59 2.38 2.38 0 00-.48 1.56v8.7a2.38 2.38 0 00.48 1.56 1.55 1.55 0 001.27.59 1.61 1.61 0 001.31-.55zM458.76 286.24a12.43 12.43 0 01-.1 1.75 3.84 3.84 0 01-1.3 2.85 4.62 4.62 0 01-3.12 1 4.5 4.5 0 01-3-.95 4.19 4.19 0 01-1.4-2.85l-.08-.7a.27.27 0 01.3-.3h2.3a.3.3 0 01.33.3 2.32 2.32 0 00.05.45c.2 1 .7 1.55 1.5 1.55a1.34 1.34 0 001-.4 2 2 0 00.51-1.12 10.08 10.08 0 00.1-1.6 7.44 7.44 0 00-.15-1.6 1.8 1.8 0 00-.51-1 1.36 1.36 0 00-1-.36 1.49 1.49 0 00-1.55 1.15.32.32 0 01-.33.28H450a.26.26 0 01-.21-.09.29.29 0 01-.09-.21v-9.93a.29.29 0 01.09-.21.26.26 0 01.21-.09h8.15a.31.31 0 01.3.3v1.9a.26.26 0 01-.09.21.29.29 0 01-.21.09h-5.43c-.08 0-.12 0-.12.13v4.87c0 .05 0 .08.05.09s.06 0 .1 0a3 3 0 011-.71 3.09 3.09 0 011.28-.26 3.19 3.19 0 012.42 1 4.48 4.48 0 011.08 2.71 10 10 0 01.23 1.75zM461.44 290.58a4.87 4.87 0 01-1.28-3.51v-8.35a4.91 4.91 0 011.28-3.52 4.44 4.44 0 013.37-1.33 4.53 4.53 0 013.42 1.33 4.91 4.91 0 011.28 3.52v8.35a4.87 4.87 0 01-1.28 3.51 4.54 4.54 0 01-3.42 1.34 4.44 4.44 0 01-3.37-1.34zm4.69-1.78a2.38 2.38 0 00.48-1.56v-8.7a2.38 2.38 0 00-.48-1.56 1.65 1.65 0 00-1.32-.59 1.55 1.55 0 00-1.27.59 2.38 2.38 0 00-.48 1.56v8.7a2.38 2.38 0 00.48 1.56 1.52 1.52 0 001.27.59 1.62 1.62 0 001.32-.59zM461.09 532.63a4.87 4.87 0 01-1.27-3.51v-8.35a4.87 4.87 0 011.27-3.51 4.47 4.47 0 013.38-1.34 4.52 4.52 0 013.41 1.34 4.84 4.84 0 011.29 3.51v8.35a4.84 4.84 0 01-1.29 3.51 4.48 4.48 0 01-3.41 1.34 4.43 4.43 0 01-3.38-1.34zm4.69-1.77a2.33 2.33 0 00.49-1.56v-8.7a2.38 2.38 0 00-.49-1.57 1.6 1.6 0 00-1.31-.58 1.57 1.57 0 00-1.28.58 2.49 2.49 0 00-.47 1.57v8.7a2.44 2.44 0 00.47 1.56 1.58 1.58 0 001.28.59 1.61 1.61 0 001.31-.59z"
        />
      </g>
    </g>
  </svg>
</div>
`;
